src/logic/referral/referral.ts

```diff
@@ -260,6 +321,18 @@ export async function createReferralComponent(
 
       const { status: currentStatus, referrer } = progress[0]
 
+      const cacheKey = `referral:invited-user:${invitedUser}`
+      const cachedInvitedUserLogins: string[] = (await redis.get(cacheKey)) || []
+
+      if (cachedInvitedUserLogins.length < REFERRAL_MIN_LOGIN_DAYS) {
+        const loginDays = new Set(cachedInvitedUserLogins)
+        loginDays.add(new Date().toISOString().split('T')[0])
+        await redis.put(cacheKey, Array.from(loginDays), { noTTL: true })
+        throw new ReferralInvalidInputError(`User must have logged in at least ${REFERRAL_MIN_LOGIN_DAYS} days`)
```

Comment:
doubt, throwing here is expected? it won't break any flow right? if it will maybe we could just return